0_global_functions.R

```diff
@@ -21,310 +21,289 @@ is_blank_na <- function(x){
   flag
 }
 
-read_file <- function(file_name){
-  # This function checks the file type and reads it in. 
-  # This should allow switching between fst, rda and csv to happen more simply
-  
-  extension_type = substring(file_name,nchar(file_name)-4+1)
-  if (!extension_type %in% c(".rda", ".csv", ".fst")){stop("Can't read in file. File extension type not applicable.")}  
-  
-  df <- NA
-  
-  if(file.exists(file_name)){
-    
-    if (extension_type == ".rda"){df <- readr::read_rds(file_name)}
-    if (extension_type == ".fst"){df <- fst::read_fst(file_name)}
-    if (extension_type == ".csv"){df <- readr::read_csv(file_name)}
-  }else{
-    warning(paste0(file_name, " does not exist"))
-  }
-  
-  return(df)
-}
-
 save_file <- function(df, file_name){
-  
-  
+
+
   extension_type = substring(file_name,nchar(file_name)-4+1)
-  if (!extension_type %in% c(".rda", ".csv", ".fst")){stop("Can't read in file. File extension type not applicable.")}  
-  
+  if (!extension_type %in% c(".rda", ".csv", ".fst")){stop("Can't read in file. File extension type not applicable.")}
```

Comment:
Why is the `save_file()` function reporting that it "Can't read in file."?

Reply:
I didn't work on this function but it seems similar to the one I did work on. I'll open an issue to follow up. Hopefully we can reuse some of the code of this PR.

In any case, I'm now sure if it pays of to maintain this wrappers around functions that read and save files. The "convenience" they add comes at the cost of having to maintain too many lines of code and at the risk of introducing lots of bugs. 

I'm not attached to this PR and I would love to kill `read_file()` if and whenever it is safe.